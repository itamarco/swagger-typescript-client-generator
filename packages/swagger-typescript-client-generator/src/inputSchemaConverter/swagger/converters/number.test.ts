import { convertSwaggerSchemaToNormalizedSchema } from "./index"
import * as Swagger from "swagger-schema-official"

const testCases: Swagger.Schema[] = [
  {
    type: "number",
    description: "type number"
  },
  {
    type: "integer",
    description: "type integer"
  },
  {
    type: "number",
    minimum: 1,
    description: "type and numeric minimum"
  },
  {
    type: "number",
    minimum: ("1" as unknown) as number,
    description: "type and string minimum"
  },
  {
    type: "number",
    maximum: 1,
    description: "type and numeric maximum"
  },
  {
    type: "number",
    maximum: ("1" as unknown) as number,
    description: "type and string maximum"
  },
  {
    type: "number",
    default: 1,
    description: "type and numeric default"
  },
  {
    type: "number",
    default: ("1" as unknown) as number,
    description: "type and string default"
  },
  {
    type: "number",
    format: "timestamp",
    description: "type and format"
  },
  {
    type: "number",
    enum: [],
    description: "type and empty enum"
  },
  {
    type: "number",
    enum: [1, 2, 3],
    description: "type and numeric enum"
  },
  {
    type: "number",
    enum: [true, "2", 3],
    description: "type and mixed enum"
  }
]

const type = "number"

test.each(testCases.map(item => [item.description, item]))(
  `converter converts scenario "%s"`,
  (description, definition) => {
    const spec = convertSwaggerSchemaToNormalizedSchema(definition as unknown)
    expect(spec.type).toEqual(type)
    expect(spec).toMatchSnapshot()
  }
)
