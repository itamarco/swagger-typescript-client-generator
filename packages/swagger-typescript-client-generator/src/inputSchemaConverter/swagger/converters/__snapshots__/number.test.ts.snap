// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter converts scenario "type and empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and empty enum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "format": "timestamp",
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and numeric default" 1`] = `
Object {
  "default": 1,
  "description": "type and numeric default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and numeric enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric enum",
  "enum": Array [
    1,
    2,
    3,
  ],
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and numeric maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": 1,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and numeric minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and numeric minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": 1,
  "type": "number",
}
`;

exports[`converter converts scenario "type and string default" 1`] = `
Object {
  "default": 1,
  "description": "type and string default",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and string maximum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string maximum",
  "enum": undefined,
  "format": undefined,
  "maximum": 1,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type and string minimum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string minimum",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": 1,
  "type": "number",
}
`;

exports[`converter converts scenario "type integer" 1`] = `
Object {
  "default": undefined,
  "description": "type integer",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;

exports[`converter converts scenario "type number" 1`] = `
Object {
  "default": undefined,
  "description": "type number",
  "enum": undefined,
  "format": undefined,
  "maximum": undefined,
  "minimum": undefined,
  "type": "number",
}
`;
